pkg/armhelpers/azureclient.go

```diff
@@ -462,3 +463,45 @@ func (az *AzureClient) addAcceptLanguages() autorest.PrepareDecorator {
 		})
 	}
 }
+
+func (az *AzureClient) addAuxiliaryTokens() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			r, err := p.Prepare(r)
+			if err != nil {
+				return r, err
+			}
+			if az.auxiliaryTokens != nil {
+				for _, token := range az.auxiliaryTokens {
+					if token == "" {
+						continue
+					}
+
+					r.Header.Set("x-ms-authorization-auxiliary", fmt.Sprintf("Bearer %s", token))
+				}
+			}
+			return r, nil
+		})
+	}
+}
+
+// AddAuxillaryTokens sets the list of aux tokens to accept on this request
+func (az *AzureClient) AddAuxillaryTokens(tokens []string) {
+	az.auxiliaryTokens = tokens
+	az.authorizationClient.Client.RequestInspector = az.addAuxiliaryTokens()
```

Comment:
We seem to be calling az.addAuxiliaryTokens() multiple times only to get the same instance of autorest.PrepareDecorator. We could assign this to a variable instead and re-use that.

Reply:
Aux token is being assigned to `az.auxiliaryTokens` and then set in the Header `addAuxiliaryTokens`. I am following the coding convention in the file used for `addAcceptLanguages`